switchlink/switchlink_link_test.cc

```diff
@@ -36,79 +36,96 @@ extern "C" {
     sequentially, not in parallel, or (2) each test case is run in a
     separate process. If that's not the case, we will need to come up
     with a more creative solution (such as thread-specific storage).
+
+    TODO: Look into using CMock instead of the clumsy methodology used
+    in this test.
 ******************************************************************************/
 
 enum handler_type {
-  CREATE_INTERFACE = 1,
-  DELETE_INTERFACE = 2,
-  CREATE_TUNNEL_INTERFACE = 3,
-  DELETE_TUNNEL_INTERFACE = 4,
+  CREATE_INTERFACE = 1,           // switchlink_create_interface
+  DELETE_INTERFACE = 2,           // switchlink_delete_interface
+  CREATE_TUNNEL_INTERFACE = 3,    // switchlink_create_tunnel_interface
+  DELETE_TUNNEL_INTERFACE = 4,    // switchlink_delete_tunnel_interface
+  CREATE_VRF = 5,                 // switchlink_create_vrf
```

Comment:
Needs a `TEST_F`  for this case, which is not present in the file.
Please add a `TODO` to test this case

Reply:
No, it doesn't. The UUT never calls `switchlink_create_vrf()`. We have to provide a defining instance of the function because a different function in the same source file depends on it, and we get a link error if we don't provide a definition.

I have returned the function definition to its original implementation, in which it simply returns zero. I have also rewritten the function comment to state explicitly that the function is there to fulfill a link-time requirement and that the UUT will never call it.